src/extension.ts

```diff
@@ -199,6 +202,69 @@ export async function activate(context: vscode.ExtensionContext, perfStats: { lo
     ext.managedIdentityBranchDataProvider = new ManagedIdentityBranchDataProvider();
     ext.v2.api.resources.registerAzureResourceBranchDataProvider(AzExtResourceType.ManagedIdentityUserAssignedIdentities, ext.managedIdentityBranchDataProvider);
 
+
+    const AUTH_PROVIDER_ID = 'microsoft'; // VS Code Azure auth provider
+    const SCOPES = ['https://management.azure.com/.default']; // Default ARM scope
+
+    registerEvent(
+        'treeView.onDidChangeSessions',
+        vscode.authentication.onDidChangeSessions,
+        async (context: IActionContext, _e: vscode.AuthenticationSessionsChangeEvent) => {
```

Comment:
Please factor this into a method and make sure to call that method once in addition to registering it as a callback. `vscode.authentication.onDidChangeSessions` is not guaranteed to be called on activation.

Reply:
done.